pkg/apis/autoscaling/v1alpha1/kpa_types.go

```diff
@@ -164,18 +173,22 @@ func (rs *PodAutoscalerStatus) MarkInactive(reason, message string) {
 	podCondSet.Manage(rs).MarkFalse(PodAutoscalerConditionActive, reason, message)
 }
 
-// CanScaleToZero checks whether the pod autoscaler has been in an inactive state
-// for at least the specified grace period.
-func (rs *PodAutoscalerStatus) CanScaleToZero(gracePeriod time.Duration) bool {
+func (rs *PodAutoscalerStatus) CanScaleToZero(idlePeriod, gracePeriod time.Duration) (bool, error) {
 	if cond := rs.GetCondition(PodAutoscalerConditionActive); cond != nil {
 		switch cond.Status {
+		case corev1.ConditionTrue:
+			// Return an error if this PodAutoscaler has not been active for
+			// at least the idle period
+			if !time.Now().After(cond.LastTransitionTime.Inner.Add(idlePeriod)) {
+				return false, fmt.Errorf("Target must be active for atleast scale-to-zero idle period: %ds", idlePeriod/time.Second)
```

Comment:
I'm not sure that returning `false` would be quite correct, since we wouldn't want to return `true` if it was ConditionTrue, regardless of the LTT.

I've split this out into a separate function, `CanMarkInactive`, to handle the Active -> Inactive transition independently from Inactive -> Scale-to-Zero.